### PR TITLE
Fix percentComplete should be a float not an int

### PR DESCRIPTION
**Pull request by timotheeguerin**
fix [#832](https://github.com/Azure/typespec-azure/issues/832)

### CODE REVIEW

File: packages/typespec-azure-resource-manager/lib/common-types/types.tsp [hunk 1]
```diff
@@ -217,7 +217,7 @@ model OperationStatusResult {
   @doc("Percent of the operation that is complete.")
   @minValue(0)
   @maxValue(100)
-  percentComplete?: int32;
+  percentComplete?: float64;
```
Comment (timotheeguerin) [inline packages/typespec-azure-resource-manager/lib/common-types/types.tsp:220]: In the original issue it was mentioning `float32` but we also reference `float64` here https://github.com/Azure/typespec-azure/blob/a72d7efeacb6e6b8d64b9c65853ecc42558d77d5/packages/typespec-azure-resource-manager/lib/responses.tsp#L124

So probably good those 2 are consistent, so do we need to change the other to float32 or this one is good.

Comment (allenjzhang) [reply packages/typespec-azure-resource-manager/lib/common-types/types.tsp:220]: https://github.com/Azure/azure-resource-manager-rpc/blob/8231d7df51aec87e6ccb5bcc04478136758b0a4c/v1.0/async-api-reference.md?plain=1#L164

Comment (timotheeguerin) [reply packages/typespec-azure-resource-manager/lib/common-types/types.tsp:220]: so sound like this fix is good then

**Review by markcowl — APPROVED** [review]

File: packages/typespec-azure-resource-manager/lib/common-types/types.tsp (hunk 1 above)
Comment (markcowl) [reply packages/typespec-azure-resource-manager/lib/common-types/types.tsp:220]: Yes, double and float64 are synonymous

**Pull request merged**